### PR TITLE
Reply to comment

### PR DESCRIPTION
**Pull request by dmardassi**
Add TC for reply to comment feature.

### CODE REVIEW

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java [hunk 1]
```diff
@@ -0,0 +1,1293 @@
...
+        manageLogInOut.signIn(DATA_USER2, PLFData.DATA_PASS);
+        homePagePlatform.goToConnections();
+        connectionsManagement.acceptAConnection(DATA_USER1);
+        homePagePlatform.goToHomePage();
+        activityStream.commentActivity(activity1, comment);
+        manageLogInOut.signOut();
+        manageLogInOut.signInCas(PLFData.DATA_USER1, "gtngtn");
+        homePagePlatform.goToHomePage();
+        activityStream.replytocomment(comment, reply, DATA_NAME_USER1);
+        refresh();
```
Comment (thomasdelhomenie) [inline platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java]: Shouldn't we check if the reply exists and has the right content ?

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java [hunk 2]
```diff
@@ -0,0 +1,1293 @@
...
+        executeJavaScript("window.scrollBy(0,-2000)", "");
+        activityStream.deleteactivity(activity1);
+        executeJavaScript("window.scrollBy(0,-2000)", "");
+        homePagePlatform.goToConnections();
+        connectionsManagement.removeConnection(DATA_USER2);
+    }
+
+    @Test
+    public void test02_DeleteReplyToCommentInAS() {
```
Comment (thomasdelhomenie) [inline platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java]: It looks like the same test than test01_ReplyToCommentInAS, no ?

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java [hunk 3]
```diff
@@ -98,6 +104,65 @@ public void deleteTask(String task) {
     ELEMENT_TASK_BUTTON_DELETE_OK.click();
   }
 
+  public void addCowroker(String task) {
```
Comment (thomasdelhomenie) [inline task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: addCowroker -> addCoworker

Comment (thomasdelhomenie) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Parameter names should be more meaningful. In this case, I think it is the text of the task, so it should be taskText or taskContent.
This remarks valid for all the others methods.

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java [hunk 4]
```diff
@@ -98,6 +104,65 @@ public void deleteTask(String task) {
...
+  public void commentTask(String task, String comment) {
+    $(byText(task)).click();
+    SelenideElement frame= ELEMENT_INPUT_COMMENT_TASK;
+    switchTo().frame(frame);
+    COMMENT_INPUT_AREA.sendKeys(comment);
+    switchTo().defaultContent();
+    COMMENT_BUTTON.click();
+  }
+
+  public void deletecomment (String task, String comment) {
```
Comment (thomasdelhomenie) [inline task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: deletecomment -> deleteComment

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java [hunk 5]
```diff
@@ -98,6 +104,65 @@ public void deleteTask(String task) {
...
+    $(byText(task)).click();
+    $(byText(comment)).hover();
+    $(byText(comment)).parent().parent().parent().find(byClassName("uiIconTrashMini")).click();
+    refresh();
+    $(byText(comment)).shouldNot(Condition.exist);
+  }
+
+  public void deletereply (String task, String reply){
```
Comment (thomasdelhomenie) [inline task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: deletereply -> deleteReply

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java [hunk 6]
```diff
@@ -98,6 +104,65 @@ public void deleteTask(String task) {
...
+    $(byText(task)).click();
+    $(byText(reply)).hover();
+    $(byText(reply)).parent().parent().find(byClassName("uiIconTrashMini")).click();
+    $(byText(reply)).shouldNot(Condition.exist);
+  }
+  public void replytocommentTask(String task, String reply, String user) {
```
Comment (thomasdelhomenie) [inline task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: replytocommentTask -> replyToTaskComment

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java [hunk 7]
```diff
@@ -98,6 +104,65 @@ public void deleteTask(String task) {
...
+    ELEMENT_LABEL_REPLY_TASK.click();
+    SelenideElement frame= ELEMENT_INPUT_COMMENT_TASK;
+    switchTo().frame(frame);
+    COMMENT_INPUT_AREA.sendKeys(reply);
+    switchTo().defaultContent();
+    COMMENT_BUTTON.click();
+    $(byText(reply)).parent().parent().parent().parent().find(byText(user)).should(Condition.exist);
+  }
+
+  public void replytoreply(String task,String reply,String replytoreply, String user) {
```
Comment (thomasdelhomenie) [inline task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: replytoreply -> replyToReply

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java [hunk 8]
```diff
@@ -98,6 +104,65 @@ public void deleteTask(String task) {
...
+    $(byText(reply)).parent().parent().find(byClassName("replyCommentLink")).click();
+    SelenideElement frame= ELEMENT_INPUT_COMMENT_TASK;
+    switchTo().frame(frame);
+    COMMENT_INPUT_AREA.sendKeys(replytoreply);
+    switchTo().defaultContent();
+    COMMENT_BUTTON.click();
+    $(byText(replytoreply)).parent().parent().parent().parent().find(byText(user)).should(Condition.exist);
+  }
+
+  public void showallreplies(String task, String comment) {
```
Comment (thomasdelhomenie) [inline task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: showallreplies -> showAllReplies

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java [hunk 9]
```diff
@@ -98,6 +104,65 @@ public void deleteTask(String task) {
...
+    $(byText(task)).click();
+    String idDataComment = $(byText(comment)).parent().parent().getAttribute("data-commentid");
+    // Get id Comment button
+    $(byId(ELEMENT_VIEW_ALL_REPLIES_LINK_TASK.replace("{id}", idDataComment))).waitUntil(Condition.appears, Configuration.timeout).findElementByClassName("subCommentShowAllLink").click();
+  }
   /**
```
Comment (thomasdelhomenie) [inline task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Add blank line between methods.

File: social/src/test/java/org/exoplatform/platform/qa/ui/social/smoke/SOCHomePageTestIT.java [hunk 10]
```diff
@@ -1,26 +1,25 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
 import static com.codeborne.selenide.Selectors.*;
 import static com.codeborne.selenide.Selenide.$;
 import static com.codeborne.selenide.Selenide.executeJavaScript;
 import static org.exoplatform.platform.qa.ui.selenium.Utils.getRandomNumber;
 import static org.exoplatform.platform.qa.ui.selenium.locator.ActivityStreamLocator.*;
+import static org.exoplatform.platform.qa.ui.selenium.locator.HomePageLocator.ELEMENT_COMMENT_BLOC;
```
Comment (thomasdelhomenie) [inline social/src/test/java/org/exoplatform/platform/qa/ui/social/smoke/SOCHomePageTestIT.java]: Useless ?

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/platform/ActivityStream.java [hunk 11]
```diff
@@ -1033,4 +1056,195 @@ public void likeUnlikeComment(String activity, String comment) {
   public enum changeTypes {
     No_Value, Has_One_Value;
   }
+  public void replytocomment(String comment, String reply,String user) {
```
Comment (thomasdelhomenie) [inline selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/platform/ActivityStream.java]: replytocomment -> replyToComment
Also please rename all others methods correctly.

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/taskmanagement/TaskManagementLocator.java [hunk 12]
```diff
@@ -164,6 +163,16 @@
                                                                      $(byId("taskManagement")).waitUntil(Condition.appears,
                                                                                                          Configuration.timeout)
                                                                                               .find(byXpath("//*[@id=\"taskManagement\"]/div[1]/ul/li[3]/div[2]/ul/li[1]/div/div/ul/li[3]/a"));
+public static final SelenideElement ELEMENT_INPUT_COMMENT_TASK=$(byClassName("rightPanelContent")).find(byClassName("cke_wysiwyg_frame"));
```
Comment (thomasdelhomenie) [inline selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/taskmanagement/TaskManagementLocator.java]: Indentation is not good.

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java (hunk 1 above)
Comment (dmardassi) [reply platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java]: I checked it in the method replytocomment()

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java (hunk 2 above)
Comment (dmardassi) [reply platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/ReplyToCommentTestIT.java]: The difference is that we delete the reply after adding it using this method deletereplyinAS()

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java (hunk 9 above)
Comment (dmardassi) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Done

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java (hunk 8 above)
Comment (dmardassi) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Done

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java (hunk 7 above)
Comment (dmardassi) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Done

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java (hunk 6 above)
Comment (dmardassi) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Done

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java (hunk 5 above)
Comment (dmardassi) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Done

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java (hunk 4 above)
Comment (dmardassi) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Done

File: task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java (hunk 3 above)
Comment (dmardassi) [reply task/src/main/java/org/exoplatform/platform/qa/ui/task/pageobject/TasksManagement.java]: Done

File: social/src/test/java/org/exoplatform/platform/qa/ui/social/smoke/SOCHomePageTestIT.java (hunk 10 above)
Comment (dmardassi) [reply social/src/test/java/org/exoplatform/platform/qa/ui/social/smoke/SOCHomePageTestIT.java]: yes, I deleted it.

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/platform/ActivityStream.java (hunk 11 above)
Comment (dmardassi) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/platform/ActivityStream.java]: Done

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/taskmanagement/TaskManagementLocator.java (hunk 12 above)
Comment (dmardassi) [reply selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/taskmanagement/TaskManagementLocator.java]: Done

**Pull request merged**